### DISCUSSION TITLE
Is it possible to see which inputs / refs / errors appear first on a page?

### DISCUSSION THREAD

**Discussion by jvgeee** [react-hook-form/react-hook-form#7714] [Q&A]
Using React Native, I have a task where I'd like to scroll to the first error on a page when triggering a form submission.

When I submit, I get errors, like so

```
{"givenName": {"message": "Please enter your First name", "ref": {"name": "givenName"}, "type": "required"}, "surname": {"message": "Please enter your Last name", "ref": {"name": "surname"}, "type": "required"}}
```

I'd like to programmatically find which of these inputs is rendered first on the page; potentially by finding which of these inputs has the lowest pageY value, and then scroll to it.

Are there any helper methods in RHF that might be useful here? or is it possible to access the ref on each input and do something like:

```

        nameRef?.current?.measure((x, y, width, height, pageX, pageY) =>  pageY ))

```

Or is this something that I pretty much have to completely out of RHF? 

EDIT: I see that if I add the `field.ref` ref from useController I get a `ref.focus` function on the error, is there a way of dynamically seeing which of these appears first on the page?

**Comment by Moshyfawn** [comment]
As you said, the `field.ref` helps manage the input focus. There's no way to know the order of the inputs on the page as they can be dynamically hidden, for example. The easiest way to keep the order of the fields is to use [@hookform/resolvers](https://github.com/react-hook-form/resolvers). See #7523 discussion for more context.
